### ISSUE TITLE
Ditch our tv_input_api wrapper around presubmit api

### ISSUE THREAD

**Issue by natduca** [google/trace-viewer#1059]
We wrap depot tools' input_api in a tv_input_api, in build/tv_input_api.py

We should ditch it. @petrcermak

**Comment by petrcermak** [comment]
I totally agree. I'm happy to do this, but it might take some time before I get to it.

**Comment by natduca** [comment]
@anniesullie this probably blocks catapult right? given that @petrcermak is a bit overloaded, maybe we need to find alternate folks to pick itu p

**Comment by petrcermak** [comment]
I looked through the wrapper code and the situation is as follows:

  * <tt>class **_AffectedFile**</tt> (wrapper of depot_tools [<tt>AffectedFile</tt>](https://code.google.com/p/chromium/codesearch#chromium/tools/depot_tools/presubmit_support.py&l=574))
      * **Most properties/methods** (<tt>\__repr\__()</tt>, <tt>filename</tt>, <tt>absolute_path</tt>, <tt>contents_as_lines</tt>, and <tt>changed_lines</tt>) map directly to properties of <tt>AffectedFile</tt> (<tt>filename</tt>, <tt>LocalPath()</tt>, <tt>AbsoluteLocalPath()</tt>, <tt>NewContents()</tt>, and <tt>ChangedContents()</tt>). Therefore, it should be very **straightforward to unwrap**.
      * **<tt>contents</tt> property** is just a simple list join (i.e. <tt>f.contents = '\n'.join(f.contents_as_lines)</tt>), so it would be straightforward to unwrap as well, but it also **adds caching**. We need to check how much slower the pre-submit hook would become if we removed the caching.

  * <tt>class **TvInputAPI**</tt> (wrapper of depot_tools [<tt>InputApi</tt>](https://code.google.com/p/chromium/codesearch#chromium/tools/depot_tools/presubmit_support.py&l=220))

    * **<tt>AffectedFiles()</tt> method** can be **simply removed** once the <tt>_AffectedFile</tt> wrapper above is removed (there's an identical method in depot_tools <tt>InputApi</tt>).
    * **<tt>IsIgnoredFile()</tt> method** is used only by the <tt>JSChecker</tt> class, so it could be moved there. A more future-proof solution would be to **move it to the <tt>TracingProject</tt> class**.
    * **<tt>IsThirdParty()</tt> method** is used in at least two different places, so it would probably make the most sense to **move it to the <tt>TracingProject</tt> class**.

To sum up, the following changes need to be done:

  1. Unwrap method calls and property accesses of <tt>\_InputApi</tt> (very straightforward).
  2. Check the performance impact of removing <tt>_InputApi.contents</tt> caching.
  3. Move <tt>TvInputAPI.IsIgnoredFile()</tt> and <tt>TvInputAPI</tt> to <tt>TracingProject</tt>.

Sounds reasonable?

**Comment by petrcermak** [comment]
Given that this seemed to be a blocker, I moved it up my priority list, and the *wrapper is now removed*. One last thing before we close this issue — **do we need to have two pre-submit scripts (_PRESUBMIT.py_ and _tracing/PRESUBMIT.py_)?**

**Comment by dj2** [comment]
I think we do for now. The top level one will become a catapult presubmit and we may have a tracing specific presubmit.

We're just in the weird, half-catapult space at the moment.

**Comment by anniesullie** [comment]
thanks so much, @petrcermak! +1 to what @dj2 said--the top-level one will be merged with catapult's, and there may be a tracing-specific one.

**Comment by petrcermak** [comment]
In that case, this is now **closed** :blush:
